source/slang/slang-emit-c-like.cpp

```diff
@@ -2774,6 +2774,26 @@ void CLikeSourceEmitter::defaultEmitInstExpr(IRInst* inst, const EmitOpInfo& inO
     {
         break; //should already have been parsed and used.
     }
+    case kIROp_StaticAssert:
```

Comment:
Adding a check here is likely too late. C-like emit also does not cover spirv backend.

We need a separate pass to do the static assert checking.

The problem with this kind of post link checking is that it introduces situations where the code checked fine in the front end, but fails after linking. This usually leads to user frustrations and the general design philosophy is to avoid them when possible and try our best to capture errors as early as possible.

For this particular case the only way to allow front end check is to duplicate all the sampling functions, but that creates a lot of bloat in the stdlib, so doing the post link check may be the right trade-off.

Reply:
I appreciate this feedback.
Can I ask more help on where the checking should happen?
I am thinking maybe I can do it in `generateIRForTranslationUnit()` where the missing-return check and operator<< check happens.

Reply:
This likely need to happen late because we have to inline and specialize our generics before we can resolve those constexpr values. We can just do it right before going into emit at the end of linkAndOptimizeIR.